guides/release/templates/handlebars-basics.md

```diff
@@ -1,85 +1,203 @@
 
-### Helpers
+```javascript {data-filename=app/components/my-component.js}
+import Component from '@ember/component';
 
-Ember Helpers are functions that can compute values and can be used in any template.
+export default Component.extend({
+  firstName: 'Jessica',
+  lastName: 'Jordan'
+});
+```
 
-Ember gives you the ability to [write your own helpers](../writing-helpers/), to bring a minimum of logic into Ember templating.
+```handlebars {data-filename=app/templates/components/my-component.hbs}
+Hello, <strong>{{this.firstName}} {{this.lastName}}</strong>!
+```
+
+This would render:
+
+```html
+Hello, <strong>Jessica Jordan</strong>!
+```
+
+## Things you might see in a Template
+
+A lot more than just HTML markup can go in Templates. In the other pages of this guide, we will cover the features one at a time. In general, special Ember functionality will appear inside curly braces, like this: `{{example}}`. Here are a few examples of Ember Handlebars in action:
+
+Route example:
+```hbs {data-filename=app/templates/application.hbs data-update=true}
+{{!-- Example of a comment that will be invisible --}}
+
+{{!-- outlet determines where a child route's content should render. Don't delete it until you know more about it! --}}
+<div>
+  {{outlet}}
+</div>
+
+{{!-- One way to use a component within a template. This is called "Angle Bracket Component syntax." --}}
+<MyComponent />
+
+```
+
+Component example:
+```hbs {data-filename=app/components/templates/my-component.hbs data-update=true}
+{{!-- An attribute that is defined in a component's JavaScript file --}}
+{{this.numberOfSquirrels}}
+
+{{!-- Some data passed down from a parent component or controller --}}
+{{weatherStatus}}
+
+{{!-- This button uses Ember Actions to make it interactive. A method named `plantATree` is called when the button is clicked --}}
+<button onclick={{action 'plantATree'}}>
+  More trees!
+<button>
+
+{{!-- Here's an example of template logic in action. If the `this.skyIsBlue` attribute is `true`, the text inside will be shown --}}
+{{#if this.skyIsBlue}}
+  If the skyIsBlue attribute is true, show this message
+{{/if}}
+
+{{!-- You can pass a whole block of markup and handlebars content from one component to another. yield is where the block shows up when the page is rendered --}}
+{{yield}}
+```
+
+Lastly, it's important to know that data can be passed from
+one Template to another. Here's what it looks like
+to pass data from a Route Template to a Component Template.
+
+```handlebars {data-filename=app/templates/application.hbs}
+<MyComponent @favoriteFramework=this.favoriteFramework />
+```
+
+Learn more about passing data between Templates in [../../components/passing-properties-to-a-component].
+
+## Helper functions
+
+Ember Helpers are pure functions that can be used in any template to change how data is displayed. For example, you could write a Helper function that capitalizes a word, does some math, converts a currency, or more.
```

Comment:
I feel like this undersells helpers a bit. They can be used for data display, but they can also contain logic and data flow. `{{if}}` and `{{each}}` are "helper" ish, for instance, and I think it's important for us to mention this. 

One place people really get frustrated with Ember in my experience is the lack for interpolating JavaScript, like JSX. Helpers _do_ allow you to do this, but in a controlled way that prevents you from putting _too_ much logic in your template. It's really LISP-y too, which encourages functional patterns! 

I would personally approach this as "Ember let's you do what you want", so as not to upset anyone who is a fan of "logic-less templates", but also show that we are _enabling_ the patterns that work for _you_ 😄

Reply:
Replaced first sentence with `Ember Helpers are a way to use JavaScript logic in your templates.`